aspnetcore/fundamentals/middleware/extensibility-third-party-container/sample/Startup.cs

```diff
@@ -0,0 +1,87 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using SimpleInjector;
+using SimpleInjector.Lifestyles;
+using MiddlewareExtensibilitySample.Data;
+using MiddlewareExtensibilitySample.Middleware;
+
+namespace MiddlewareExtensibilitySample
+{
+    public class Startup
+    {
+        private Container container = new Container();
+
+        #region snippet1
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc();
+
+            // Replace the default middleware factory with the 
+            // SimpleInjectorMiddlewareFactory. Pass in the 
+            // Simple Injector container when the factory is
+            // instantiated.
+            services.AddTransient<IMiddlewareFactory>(_ =>
+            {
+                return new SimpleInjectorMiddlewareFactory(container);
+            });
+
+            // Wraps ASP.NET requests in an 
+            // SimpleInjector.Lifestyles.AsyncScopedLifestyle.
+            services.UseSimpleInjectorAspNetRequestScoping(container);
+
+            // Provide the AppDbContext from the Simple Injector
+            // container whenever it's requested from the default
+            // service container.
+            services.AddScoped<AppDbContext>(provider => container.GetInstance<AppDbContext>());
+
+            // Sets the default scoped lifestyle that the 
+            // container should use when a registration is made 
+            // using SimpleInjector.Lifestyle.Scoped.
+            container.Options.DefaultScopedLifestyle = new AsyncScopedLifestyle();
+
+            // Register the AppDbContext in the Simple Injector
+            // container. Use an options builder to specify the
+            // use of an in-memory database.
+            container.Register<AppDbContext>(() => 
+            {
+                var optionsBuilder = new DbContextOptionsBuilder<DbContext>();
+                optionsBuilder.UseInMemoryDatabase("InMemoryDb");
+                return new AppDbContext(optionsBuilder.Options);
+            }, Lifestyle.Scoped);
+
+            // Register the middleware factory with the Simple
+            // Injector container.
+            container.Register<SimpleInjectorMiddlewareFactory>();
```

Comment:
Why do you need this?

Reply:
I prob don't need it. I'll remove and see if all is well.